### PR TITLE
Homework 6 

### PR DESCRIPTION
**Pull request by ifochka9519**
Spring Rest Task

### CODE REVIEW

File: Dump20190526.sql [hunk 1]
```diff
@@ -0,0 +1,37 @@
+-- MySQL dump 10.13  Distrib 5.7.24, for osx10.13 (x86_64)
```
Comment (inventorsoft-academy) [inline Dump20190526.sql:1]: What is the purpose of this file?

File: src/main/java/com/example/homework6/controllers/EmailController.java [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        Iterable<EmailItem> emails = emailRepo.findAll();
+        ModelAndView mv = new ModelAndView();
+        mv.setViewName("emails");
+        mv.addObject("emails", emails);
+        mv.getModel();
+        return mv;
+    }
+
+    @GetMapping(value = "/email")
+    private @ResponseBody EmailItem getEmail(Integer emailId) {
```
Comment (inventorsoft-academy) [inline src/main/java/com/example/homework6/controllers/EmailController.java:63]: How the *emailId* parameter get's populated? How does Spring MVC figure it out?

File: src/main/java/com/example/homework6/controllers/EmailController.java [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+    private ModelAndView getEmails(){
+        Iterable<EmailItem> emails = emailRepo.findAll();
+        ModelAndView mv = new ModelAndView();
+        mv.setViewName("emails");
+        mv.addObject("emails", emails);
+        mv.getModel();
+        return mv;
+    }
+
+    @GetMapping(value = "/email")
```
Comment (inventorsoft-academy) [inline src/main/java/com/example/homework6/controllers/EmailController.java:62]: Make this url mapping more specific. URL is the name. If you want to get specific email you need to provide unique name to identify it from the others. Consider, adding id as url path part

File: src/main/java/com/example/homework6/controllers/EmailController.java [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+            email.setStatusDelivery(false);
+            emailRepo.save(email);
+            return getEmails();
+        } else {
+            //TODO there is no Email in the repo with 'id'
+            return null;
+        }
+    }
+
+    @GetMapping(value = "/delete")
```
Comment (inventorsoft-academy) [inline src/main/java/com/example/homework6/controllers/EmailController.java]: When you're saying: get me this ice cream, you don't expect it to be deleted, right? :) The same rules can be applied it. If you want to delete something, be clear and use DELETE http method. Next issue is the URL. Take a look at comment above

File: src/main/java/com/example/homework6/controllers/EmailController.java [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+    private @ResponseBody String deleteEmail(Integer emailId) {
```
Comment (inventorsoft-academy) [inline src/main/java/com/example/homework6/controllers/EmailController.java:94]: How *String* value will be serialized into JSON format. JSON consist of key value pairs. What will be the key for response value?

File: src/main/java/com/example/homework6/controllers/EmailController.java [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+        if(e.isPresent()) {
+            emailRepo.delete(e.get());
+            msg = "Email deleted!";
+        } else {
+            msg = "Sorry, I can't do it...";
+        }
+        return msg;
+    }
+
+    @GetMapping(value = "/send")
```
Comment (inventorsoft-academy) [inline src/main/java/com/example/homework6/controllers/EmailController.java]: Again, you're abusing GET method. GET method is intended to be used only when you want to get data out from server. In this case POST is much better

**Review by inventorsoft-academy — CHANGES_REQUESTED** [review]

File: Dump20190526.sql (hunk 1 above)
Comment (ifochka9519) [reply Dump20190526.sql:1]: This is dump of database. I'm using database in this task.

File: src/main/java/com/example/homework6/controllers/EmailController.java (hunk 2 above)
Comment (ifochka9519) [reply src/main/java/com/example/homework6/controllers/EmailController.java:63]: I'm using Ajax here

File: src/main/java/com/example/homework6/controllers/EmailController.java (hunk 3 above)
Comment (ifochka9519) [reply src/main/java/com/example/homework6/controllers/EmailController.java:62]: I'm using this url for call modal window for editing email.

File: src/main/java/com/example/homework6/controllers/EmailController.java (hunk 5 above)
Comment (ifochka9519) [reply src/main/java/com/example/homework6/controllers/EmailController.java:94]: I'm not returning JSON. I'm returning name of view. All date I'm saving in database

**Comment by ifochka9519** [comment]
Fixed!

**Pull request closed without merging**